uber/site_sections/registration.py

```diff
@@ -653,9 +653,12 @@ def take_payment(self, session, payment_id, stripeToken):
         if message:
             raise HTTPRedirect('pay?id={}&message={}', attendee.id, message)
         else:
+            db_attendee = session.query(Attendee).filter_by(id=attendee.id).first()
+            if db_attendee:
+                attendee = db_attendee
             attendee.paid = c.HAS_PAID
             attendee.amount_paid = attendee.total_cost
-            session.merge(attendee)
+            session.add(attendee)
```

Comment:
Hmm, so I'm guessing that the reason why this fixed the problem is that there was state on the ``Attendee`` object we didn't want like maybe a department assignment not being tracked or something, and so calling ``session.merge`` was causing that to get re-applied which was why we were seeing the unique constraint error.

This suggests that this general strategy of using ``merge`` as a kind of "update or create" wasn't a good idea in the first place if it can lead to this kind of result.  Even if it happened to work for many years, we'd have been better off with an explicit check for whether the attendee already existed as we're now doing above.

Reply:
It’s due to the janky way I’m saving requested depts combined with a sessionized attendee

Reply:
Gotcha.  I'll make a note to look into this after the event when things are more relaxed just to better understand what's going on here.